tools/cldr-apps/src/main/java/org/unicode/cldr/web/CodeSortMode.java

```diff
@@ -36,13 +36,20 @@ String getName() {
     }
 
     @Override
-    Comparator<DataRow> createComparator() {
-        return comparator();
+    Comparator<DataRow> getComparator() {
+        return CollatorHelper.COMPARATOR;
     }
 
-    public static Comparator<DataRow> comparator() {
-        return new Comparator<DataRow>() {
-            final Collator myCollator = createCollator();
+    static final private class CollatorHelper {
```

Comment:
This looks better, except that each static should be in its own helper. Otherwise you create all of them when any one of them is created.

Reply:
The Comparator depends on the Collator.

Reply:
ok, makes sense